gengokit/httptransport/templates/client.go

```diff
@@ -212,12 +212,25 @@ func contextValuesToHttpHeaders(keys []string) httptransport.RequestFunc {
 	// error and attempt to decode the specific error message from the response
 	// body. Primarily useful in a client.
 	func DecodeHTTP{{$method.Name}}Response(_ context.Context, r *http.Response) (interface{}, error) {
+		buf, err := ioutil.ReadAll(r.Body)
+		if len(buf) == 0 {
+			return nil, errors.New("response http body empty")
+		}
+
+		if err != nil {
```

Comment:
I might switch the order of these checks, to check the error first. It'll be caught either way--but I don't know for certain what's returned for the buffer if an error occurs during a read: if a zero-byte buffer comes back but there was an actual error, it'll get reported as a zero-length response, and the actual error will not be reported.